We write a NUL to prompt_tmp[tot_size] later, so make sure that the destination
is part of the allocated block.
--- Linux-PAM-0.77/modules/pam_issue/pam_issue.c	2003-06-30 22:08:49.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_issue/pam_issue.c	2003-06-30 22:08:45.000000000 -0400
@@ -130,7 +130,7 @@
 	{
 	    char *prompt_tmp_tmp = prompt_tmp;
 
-	    prompt_tmp = realloc(prompt_tmp, tot_size);
+	    prompt_tmp = realloc(prompt_tmp, tot_size + 1);
 	    if (prompt_tmp == NULL) {
 		prompt_tmp = prompt_tmp_tmp;
 		retval = PAM_IGNORE;
